### PR TITLE
Add elasticsearch storage date format config.

### PR DESCRIPTION
**Pull request by sniperking1234**
Signed-off-by: Chen Zhengwei <chenzhengwei@inspur.com>

## Which problem is this PR solving?
- Resolves https://github.com/jaegertracing/jaeger/issues/1489
Jaeger can customize the es date format.

## Short description of the changes
- User can configure the date separator to change the date format.
- Default date formate is yyyy-MM-dd, it won't affect the old version.

Associated PR: https://github.com/jaegertracing/jaeger/pull/2637

### CODE REVIEW

**Comment by pavolloffay** [comment]
Depends on https://github.com/jaegertracing/jaeger/pull/2637

**Comment by sniperking1234** [comment]
@pavolloffay https://github.com/jaegertracing/jaeger/pull/2637 have been merged into the master. Would you please review the code.

File: jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java [hunk 1]
```diff
@@ -208,12 +217,23 @@ private static String prefix(String prefix) {
...
 
+  private static String datePattern(String separator) {
+    if (separator.equals("")) {
+      return "yyyyMMdd";
+    }
+    // ' is escape character in date format, we should double it here.
+    if (separator.contains("'")) {
+      separator = separator.replace("'", "''");
+    }
+    return String.format("yyyy'%s'MM'%s'dd", separator, separator);
```
Comment (pavolloffay) [inline jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java:228]: why there is `'%s` the `'s` ?

Comment (sniperking1234) [reply jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java:228]: The separator may be more than one character.

**Comment by jpkrohling** [comment]
@pavolloffay do you have time to review this one?

**Pull request merged**